git_machete/cmd.py

```diff
@@ -266,47 +274,55 @@ def shell_escape(arg):
     return " ".join(env_repr + [cmd] + list(map(shell_escape, args)))
 
 
-def run_git(git_cmd, *args, **kwargs):
+def run_git(git_cmd: str, *args: str, **kwargs: Dict[str, str]) -> int:
     exit_code = run_cmd("git", git_cmd, *args, **kwargs)
     if not kwargs.get("allow_non_zero") and exit_code != 0:
         raise MacheteException(f"`{cmd_shell_repr('git', git_cmd, *args, **kwargs)}` returned {exit_code}")
     return exit_code
 
 
-def popen_git(git_cmd, *args, **kwargs):
+def popen_git(git_cmd: str, *args: str, **kwargs: Dict[str, str]) -> str:
```

Comment:
Same here... I changed one invocation to `popen_git("version", foo="bar")`... and received error:

`git_machete/cmd.py:698: error: Argument "foo" to "popen_git" has incompatible type "str"; expected "Dict[str, str]"`

Reply:
So I guess the type of `kwargs` is really supposed to mean "type of each kwarg", not "type of the collection of all kwargs"

Reply:
> Same here... I changed one invocation to `popen_git("version", foo="bar")`... and received error:
> 
> `git_machete/cmd.py:698: error: Argument "foo" to "popen_git" has incompatible type "str"; expected "Dict[str, str]"`

And this is fine since the type is for each of the "kwargs" and we want it to be a `Dict`.
See my comment above and look close to what happens with `kwargs` and what is the type of its values later.